test/e2e/traces_test.go

```diff
@@ -257,3 +257,115 @@ func requestWithRetry(t *testing.T, testLabel string, client *http.Client, reque
 	testutil.Assert(t, false, fmt.Sprintf("%s: HTTP %d response not received within time limit", testLabel, expectedResponse))
 	return "", -1
 }
+
+func TestTraceTemplateQuery(t *testing.T) {
```

Comment:
I think would be nice to include unit tests for when defining the `read`/`readTemplate` flags (e.g. testing when both are defined, when the `readTemplate` flag has a wrong format, etc) - what do you think?

Reply:
I like to test parameter validation in unit tests rather than e2e tests.  It feels like too much CircleCI effort to spin up a copy of the Observatorium pod and verify that it refuses to start when both flags are there.

There are currently no unit tests for `main()`.  If we need flag validation, let's do it in a separate PR.

Reply:
yeah, I meant to add an unit test for that. I think since the flag validation is not happening, then I agree to add it in case flag validation is needed in the future